model/metric/payload_test.go

```diff
@@ -248,7 +248,11 @@ func TestPayloadDecode(t *testing.T) {
 			},
 		},
 	} {
-		payload, err := DecodePayload(test.input)
+		payloadInterface, err := DecodePayload(test.input)
+
+		// its ok to ignore the outcome of this cast here because
+		// we assert everything below
+		payload, _ := payloadInterface.(*Payload)
```

Comment:
Actually I'd say that this should even be a `require.NoError`, as if this conversion is not ok, the other checks don't make any sense and would just result in weird error messages.